### ISSUE TITLE
Pausing and disposing of window events

### ISSUE THREAD

**Issue by kim3er** [ailon/markerjs2#69]
TL;DR: I'd really like some way to pause `window` events when MarkerArea is not in focus, and dispose of them when no longer required.

We're making use of the `resultButtonBlockVisible` setting to hide the tick/cross UI. Our use case is that we're replacing our existing annotation tool with marker.js 2. It's working out great, with just one fundamental issue. Our users are used to our annotation tool being 'always on'. With the old tool, we triggered a final render just before save to capture final changes.

As I say, Marker is largely working for us. We trigger the `show` method on page load, and then `render` before saving. It should also noted that multiple annotations (so multiple MarkerAreas) can exist on the same page. Crucially though, annotations exist alongside form fields.

It's really the closing of the MarkerArea session that is the big issue for us. Our users are not used to it, and will forget to do it.

Why this is an issue, is really just one event listener that I can see. Users make an annotation, then make notes in a text area. They make a mistake, hit backspace, the text is deleted, but also the last marker in the annotation. This is because Marker adds a listener to the `keyup` event to `window`.

Unfortunately, that event does not make use of the `beforedelete` or `delete` events. If it did, I'd cancel the `beforedelete` event while the annotation tool wasn't in focus. So it'd be great if the `keyup` event could support at least `beforedelete` or a variation of.

Taking it a step further, noticing a few pointer events attached to `window` as well. A `play`/`pause` or `focus`/`blur` function that allowed you to temporarily disable the `window` events would be very handy.

As a final thought, I don't see any events being removed (at least using `removeEventListener`) on a `close` or `dispose` function. This could be something I've missed, but we allow user to create annotations within our own modals, and assuming I'm not mistaken, it'd be good to see those events cleaned up, as users may create many MarkerAreas in the lifetime of a single page.

I've read your contribution guide, which is why I've not submitted a pull request. For going any legal issues, I do stand ready to subject a pull request should you want me to.

**Comment by ailon** [comment]
Thank you for the detailed explanation of the issues and the reasoning behind them. I'll try to address these (hopefully today).

**Comment by ailon** [comment]
I've added `focus()` and `blur()` methods as well as `focus` and `blur` events. Also fixed the issue when `markerbeforedete` and `markerdelete` events weren't fired from keyboard. Thank you for bringing my attention to these!

Focus is automatically returned when user clicks inside the marker area. I wasn't brave enough to "blur" automatically when clicked outside the area, so, at least for now, you have to call `blur()` manually when your other object is focused.

**Comment by kim3er** [comment]
Hey @ailon, that's fantastic thank you. The `focus` and `blur` functions are working out great for us, and it looks like we'll be able to release on that basis.

We have encountered two points of friction integrating the new code, the automatic blurring that you mentioned above for which we've created a synthetic `blur` event.

The other, which I'll raise an issue for, is that it'd be really helpful if we could differentiate between button and keyboard events. So that we could cancel a delete that originated from the keyboard, while letting button events through.

Thanks again for moving on this so quickly, it really helped us out.
